.github/workflows/test.yml

```
@@ -41,3 +41,5 @@ jobs:
         run: |
             micromamba install python=${{ matrix.python-version }} pytest
             python -m pytest --verbose src/tests/
+        env:
+          PYTORCH_MPS_PREFER_METAL: 0  # disable MPS which doesn't work on macos-14 runner
```

This disables MPS for _all_ platforms in the matrix. Is that what you want, or do you want to disable it only for macos-14?

---

MPS is only available on macos-14, and I've added an if-condition in the unit tests to only disable MPS on devices with MPS, so should be ok? Open to a better way of doing things though.

---

Ah, I see. I suppose the comment may have thrown me off. Here, "doesn't work" doesn't mean that we can't choose to use it, but rather that when we do choose to use it, the test fails due to running out of memory. So it would work if we had enough memory.

---

Ok, maybe I can reword it a little bit, how about this:

```suggestion
        env:
          PYTORCH_MPS_PREFER_METAL: 0  # disable MPS which runs out of memory on macos-14 runner
```

We disable MPS because it runs out of memory (or rather, it doesn't allow allocating GPU-like MPS memory?) on the `macos-14` GitHub Actions runner as mentioned at https://github.com/actions/runner-images/issues/9254#issuecomment-1936326374.

---

Ok, reworded at 745110410c29e96ec1400022c6e56082906669ef